packages/sources/tradingeconomics-test/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,234 @@
+import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
+import { Server } from 'mock-socket'
+import { AddressInfo } from 'net'
+import request, { SuperTest, Test } from 'supertest'
+import { mockResponseSuccess } from './fixtures'
+import {
+  createAdapter,
+  mockStockWebSocketServer,
+  mockWebSocketProvider,
+  mockWebSocketServer,
+  setEnvVariables,
+  setupExternalAdapterTest,
+  SuiteContext,
+} from './setup'
+import { sleep } from '@chainlink/external-adapter-framework/util'
+import { WebSocketClassProvider } from '@chainlink/external-adapter-framework/transports'
+
+describe('execute', () => {
+  let spy: jest.SpyInstance
+  beforeAll(async () => {
+    const mockDate = new Date('2022-01-01T11:11:11.111Z')
+    spy = jest.spyOn(Date, 'now').mockReturnValue(mockDate.getTime())
+  })
+
+  afterAll((done) => {
+    spy.mockRestore()
+    done()
+  })
+
+  const context: SuiteContext = {
+    req: null,
+    server: async () => {
+      process.env['RATE_LIMIT_CAPACITY_SECOND'] = '5000'
+      process.env['CACHE_POLLING_MAX_RETRIES'] = '20'
+      process.env['CACHE_POLLING_SLEEP_MS'] = '500'
+      process.env['METRICS_ENABLED'] = 'false'
+      process.env['BACKGROUND_EXECUTE_MS_HTTP'] = '50'
+      const server = (await import('../../src')).server
+      return server() as Promise<ServerInstance>
+    },
+  }
+
+  const envVariables = {
+    API_CLIENT_KEY: process.env.API_CLIENT_KEY || 'fake-api-key',
+    API_CLIENT_SECRET: process.env.API_CLIENT_SECRET || 'fake-api-secret',
+  }
+
+  setupExternalAdapterTest(envVariables, context)
+
+  const id = '1'
+  describe('rest price endpoint', () => {
+    const data = {
+      id,
+      data: {
+        base: 'EUR',
+        quote: 'USD',
+      },
+    }
```

Comment:
Since this adapter is weird, could we also have a test case for the scenario the base has an overridden symbol too? Just to make sure the request transform is testing. So a request like 
```suggestion
    const data = {
      id,
      data: {
        base: 'EUR',
        quote: 'USD',
        overrides: {
            tradingeconomics: {
              EUR: "EURUSD:CUR"
            }
        },
      },
    }
```